autoscale/src/main/java/com/sequenceiq/periscope/controller/AlertController.java

```diff
@@ -176,15 +185,36 @@ private void validateAccountEntitlement(Cluster cluster) {
                 ThreadBasedUserCrnProvider.getUserCrn(),
                 ThreadBasedUserCrnProvider.getAccountId(),
                 cluster.getCloudPlatform())) {
-            throw new BadRequestException(String.format("Autoscaling is not enabled for CloudPlatform '%s', Cluster '%s' in this Account",
-                    cluster.getCloudPlatform(), cluster.getStackCrn()));
+            throw new BadRequestException(messagesService.getMessage(MessageCode.AUTOSCALING_ENTITLEMENT_NOT_ENABLED,
+                    List.of(cluster.getCloudPlatform(),  cluster.getStackName())));
+        }
+    }
+
+    private void validateSupportedHostGroup(Cluster cluster, String requestHostGroup, AlertType alertType) {
+        Set<String> supportedHostGroups = Optional.ofNullable(
+                recommendationService.getAutoscaleRecommendations(cluster.getStackCrn())).map(
+                recommendation -> {
+                    if (AlertType.LOAD.equals(alertType)) {
+                        return recommendation.getLoadBasedHostGroups();
+                    }
+                    if (AlertType.TIME.equals(alertType)) {
+                        return recommendation.getTimeBasedHostGroups();
+                    }
+                    return Set.of("");
+                }).orElse(Set.of(""));
+        if (!supportedHostGroups.contains(requestHostGroup)) {
```

Comment:
This will not NPE if there's no supported groups, correct?

Reply:
The service returns empty set when no supported group, added additional null validation.